docs/cli.rst

```diff
@@ -0,0 +1,287 @@
+Command Line Tool
+=================
+
+``anonlink-client`` includes a command line tool which can be used to interact without writing Python code.
+The primary use case is to encode personally identifiable data from a csv into Cryptographic Longterm Keys.
+
+The command line tool can be accessed in two equivalent ways:
+
+- Using the ``anonlink`` script which gets added to your path during installation.
+- directly running the python module with ``python -m anonlinkclient``.
+
+A list of valid commands can be listed with the ``--help`` argument:
+
+.. command-output:: anonlink --help
+
+
+Command specific help
+---------------------
+
+The ``anonlink`` tool has help pages for all commands built in - simply append ``--help``
+to the command.
+
+
+Hashing
+-------
+
+The command line tool ``anonlink`` can be used to hash a csv file of personally identifiable information.
+The tool needs to be provided with keys and a :ref:`schema`; it will output a file containing
+json serialized hashes.
+
+.. command-output:: anonlink hash --help
+
+
+Example
+~~~~~~~
+
+Assume a csv (``fake-pii.csv``) contains rows like the following::
+
+
+    0,Libby Slemmer,1933/09/13,F
+    1,Garold Staten,1928/11/23,M
+    2,Yaritza Edman,1972/11/30,F
+
+It can be hashed using ``anonlink`` with::
+
+    $ anonlink hash --schema simple-schema.json fake-pii.csv horse clk.json
+
+Where:
+
+- ``horse`` is the secret that both participants will use to hash their data.
+- ``simple-schema.json`` is a :ref:`schema` describing how to hash the csv. E.g, ignore the first
+  column, use bigram tokens of the name, use positional unigrams of the date of birth etc.
+- ``clk.json`` is the output file.
+
+Blocking
+--------
+The command line tool ``anonlink`` can be used to generate blocks given a csv file of personally identifiable
+information. The tool needs to be provided with keys and a :ref:`blocking-schema`; it will output a file containing
+json serialized candidate blocks.
+
+.. command-output:: anonlink block --help
+
+Example
+~~~~~~~
+
+Assume a csv (``fake-pii.csv``) contains rows like the following::
+
+
+    0,Libby Slemmer,1933/09/13,F
+    1,Garold Staten,1928/11/23,M
+    2,Yaritza Edman,1972/11/30,F
+
+It can be hashed using ``anonlink`` with::
+
+    $ anonlink block --schema blocking-schema.json fake-pii.csv horse candidate_blocks.json
+
+Describing
+----------
+
+Users can inspect the distribution of the number of bits set in ``CLKs`` by using the ``describe`` command.
```

Comment:
Will `anonlink describe` work on candidate blocks or only the clks produced by `anonlink hash`? Could be worth clarifying.